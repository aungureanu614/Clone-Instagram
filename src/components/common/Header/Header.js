import React from 'react';
import PropTypes from 'prop-types';
import { headerContainer, left, right } from './Header.module.css';

const Header = ({
  option, image, text, alt,
}) => (
  <div className={headerContainer}>
    <div className={left}>
      <img src={image} alt={alt} />
      <span>{text}</span>
    </div>
    <div className={right}>{option}</div>
  </div>
);

export default Header;

Header.propTypes = {
  option: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
