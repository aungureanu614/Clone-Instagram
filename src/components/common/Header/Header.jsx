import React from 'react'
import PropTypes from 'prop-types'
import styles from './Header.module.css'

const Header = ({ option, text, icon }) => (
  <div className={styles.headerContainer}>
    <div className={styles.left}>
      {icon}
      <span className={styles.text}>{text}</span>
    </div>
    <div className={styles.right}>{option}</div>
  </div>
)

export default Header

Header.propTypes = {
  option: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.any, // Give icon a proper type
}
