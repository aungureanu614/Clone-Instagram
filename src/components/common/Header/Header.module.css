@import url('https://fonts.googleapis.com/css?family=Pacifico');

.headerContainer {
  display: flex;
  background-color: #fff;
  border-bottom: 1px solid rgba(0,0,0,.0975);
  position: fixed;
  width: 100%;
  height: 80px;
  align-items: center;
  justify-content: space-around;
  font-family: 'Pacifico', cursive;
}

span {
  align-self: center;
  padding: 0 10px;
}

.left {
  cursor: pointer;
  display: flex;
}

.right {
   cursor: pointer;
}