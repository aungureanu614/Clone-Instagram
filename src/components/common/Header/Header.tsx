import React from 'react'
import styles from './Header.module.css'

interface Props {
  text: string
  icon: React.ReactChild
}

const Header: React.FunctionComponent<Props> = ({ icon, text }) => (
  <div className={styles.headerContainer}>
    <div className={styles.center}>
      {icon}
      <span className={styles.text}>{text}</span>
    </div>
  </div>
)

export default Header
