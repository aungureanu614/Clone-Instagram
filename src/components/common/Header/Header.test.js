import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import Header from './Header'

test('Header renders', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Header option="Sign In" text="Instagram-Clone" icon="Icon" />
    </BrowserRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
