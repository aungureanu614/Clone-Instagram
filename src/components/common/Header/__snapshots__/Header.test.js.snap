// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="headerContainer"
>
  <div
    className="center"
  >
    Icon
    <span
      className="text"
    >
      Instagram-Clone
    </span>
  </div>
</div>
`;
